### DISCUSSION TITLE
How to change / reset password without oldpassword ?

### DISCUSSION THREAD

**Discussion by yosespradipta** [pocketbase/pocketbase#5648] [Q&A] [answered]
# Resetting Staff Password Without Email Using Owner Account

## Setup

I have three collections in my PocketBase project:

1. **Owner** (auth collection): The main user type with administrative privileges.
2. **Outlet** (base collection): Owned by an Owner.
3. **Staff** (auth collection): Associated with an Outlet, can only log in using username and password (no email).

## Relationship

- An Owner has multiple Outlets.
- Each Outlet has multiple Staff members.

## Current Functionality

- Owners have full control over their Staff's login credentials.
- Owners can create and edit all Staff data.

## Problem

I can't find a way for the Owner to reset a Staff member's password. The Staff collection is an auth collection, but Staff members don't have email addresses associated with their accounts.

I have tried to simply update the password of the staff while logged in owner, but it says OldPassword cant be blank.

## Desired Outcome

I want the Owner to be able to reset any Staff member's password directly, without requiring the old password or using an email-based reset process.

## Question

What's the best way to implement this password reset functionality for Staff members, given that:
1. The reset needs to be initiated by the Owner, not the Staff member.
2. Staff accounts don't have associated email addresses.
3. We want to avoid using the old password in the reset process.

Any guidance or best practices for this scenario would be greatly appreciated. Thank you!

**Comment by ganigeorgiev** [comment, marked as answer]
Each auth collection has an extra "Manage API rule" that can be set to allow specific group of users  (including from another collection) to change directly the email, verified state, password, etc. without confirmation.

Unless I misunderstand your description, in your case you can try to go to your _`staff` collection options > API rules > Manage rule_ and set it to something like:

```js
// grant manage access to any "owner" authenticated user
@request.auth.collectionName = "owner"
```

**Reply by ganigeorgiev** [reply]
In case you want to target only the owners of specific `outlet`, then you can append to the same rule the specific related entry condition via relation/back-relation dot-notation but from your description is not clear what are the names of the `relation` fields and where they are defined. If you still need help, please export the related collections configuration as json (_Admin UI > Settings > Export collections_).

**Reply by yosespradipta** [reply]
Oh Right, the Manage API, I get it now, thanks a lot.
